File: generated-code/swift/codecraft/TcpReadWrite/Sources/TransGenTest/AttackProperties.swift
/// Entity's attack properties
public struct AttackProperties {
    /// Maximum attack range
    let attackRange: Int32

    /// Damage dealt in one tick
    let damage: Int32

    /// If true, dealing damage will collect resource from target
    let collectResource: Bool

    /// Read AttackProperties from input stream
    static func readFrom<S: InputStream>(_ stream: S) -> AttackProperties {
        var attackRange: Int32
        attackRange = stream.readInt32()
        var damage: Int32
        damage = stream.readInt32()
        var collectResource: Bool
        collectResource = stream.readBool()
        return AttackProperties(attackRange: attackRange, damage: damage, collectResource: collectResource)
    }

    /// Write AttackProperties to output stream
    func writeTo<S: OutputStream>(_ stream: S) {
        stream.writeInt32(attackRange)
        stream.writeInt32(damage)
        stream.writeBool(collectResource)
    }
}

File: generated-code/php/codecraft/FileReadWrite/Model/AttackProperties.php
<?php

namespace Model {
    require_once 'Stream.php';

    /**
     * Entity's attack properties
     */
    class AttackProperties
    {
        /**
         * Maximum attack range
         */
        public int $attackRange;
        /**
         * Damage dealt in one tick
         */
        public int $damage;
        /**
         * If true, dealing damage will collect resource from target
         */
        public bool $collectResource;
    
        function __construct(int $attackRange, int $damage, bool $collectResource)
        {
            $this->attackRange = $attackRange;
            $this->damage = $damage;
            $this->collectResource = $collectResource;
        }
    
        /**
         * Read AttackProperties from input stream
         */
        public static function readFrom(\InputStream $stream): AttackProperties
        {
            $attackRange = $stream->readInt32();
            $damage = $stream->readInt32();
            $collectResource = $stream->readBool();
            return new AttackProperties($attackRange, $damage, $collectResource);
        }
        
        /**
         * Write AttackProperties to output stream
         */
        public function writeTo(\OutputStream $stream): void
        {
            $stream->writeInt32($this->attackRange);
            $stream->writeInt32($this->damage);
            $stream->writeBool($this->collectResource);
        }
    }
}